modules/stackexchange/stackexchange.go

```diff
@@ -0,0 +1,99 @@
+package stackexchange
+
+import (
+	"github.com/laktek/Stack-on-Go/stackongo"
+	"github.com/lovelaced/glitzz/config"
+	"github.com/lovelaced/glitzz/core"
+
+	"errors"
+	"html"
+	"math/rand"
+	"strings"
+)
+
+const (
+	errInvalidSite = "Invalid site: "
+	errInvalidTags = "Tag(s) not found"
+	defaultSite    = "workplace"
+)
+
+// New registers the stackexchange module
+func New(sender core.Sender, conf config.Config) (core.Module, error) {
+	rv := &stackexchange{
+		Base: core.NewBase("stackexchange", sender, conf),
+	}
+
+	rv.AddCommand("so", rv.seStackOverflow)
+	rv.AddCommand("se", rv.seTitle)
+	rv.AddCommand("selast", rv.seLastLink)
+	return rv, nil
+}
+
+type stackexchange struct {
+	core.Base
+	lastLink string
+}
+
+func (s *stackexchange) getSite(arguments core.CommandArguments) string {
+	if len(arguments.Arguments) > 0 {
+		return arguments.Arguments[0]
+	}
+	return defaultSite
+}
+
+func (s *stackexchange) getTags(arguments core.CommandArguments) []string {
+	if len(arguments.Arguments) > 1 {
+		return arguments.Arguments[1:]
+	}
+	return nil
+}
+
+func (s *stackexchange) getRandQuestion(arguments core.CommandArguments) (*stackongo.Question, error) {
+	site := s.getSite(arguments)
+	tags := s.getTags(arguments)
+
+	session := stackongo.NewSession(site)
+	params := make(stackongo.Params)
+	params.Add("sort", "creation")
+	params.AddVectorized("tagged", tags)
+
+	questions, err := session.AllQuestions(params)
+	if err != nil {
+		if strings.Contains(err.Error(), "No site found for name") {
+			return nil, errors.New(errInvalidSite + site)
+		}
+		return nil, err
+	}
+	if len(questions.Items) == 0 {
+		return nil, errors.New(errInvalidTags)
+	}
+
+	index := rand.Intn(len(questions.Items))
+	return &questions.Items[index], nil
+}
+
+func (s *stackexchange) seTitle(arguments core.CommandArguments) ([]string, error) {
+	question, err := s.getRandQuestion(arguments)
+	if err != nil {
+		if err.Error() == errInvalidTags || strings.Contains(err.Error(), errInvalidSite) {
+			return []string{err.Error()}, nil
+		}
+		return nil, err
+	}
+
+	s.lastLink = question.Link
+	return []string{html.UnescapeString(question.Title)}, err
+}
+
+func (s *stackexchange) seLastLink(arguments core.CommandArguments) ([]string, error) {
+	if s.lastLink == "" {
+		return []string{"https://stackexchange.com"}, nil
```

Comment:
It's a fallback to avoid having the IRC module return an error on the empty string.
I could've returned an error here but since it's user induced I thought it made more sense to return a default value.

Reply:
I think it shouldn't do anything instead of returning a completely made up value. What is the point of returning a value that is equally unusable as returning no response?

Reply:
Except it will do something.
This is better than leaving the user with an error that can't be easily tracked if you ask me.